common/tls/src/main/java/io/helidon/common/tls/spi/TlsManagerProvider.java

```diff
@@ -24,4 +27,17 @@
  */
 public interface TlsManagerProvider extends ConfiguredProvider<TlsManager> {
 
+    /**
+     * Provides the ability to have a unique {@link TlsManager} per unique {@link Config} instance provided.
+     *
+     * @param configBean the config bean instance
+     * @param creator    the creator to apply if not already in cache, which takes the config bean instance
```

Comment:
We should either be very specific with this function being invoked inside synchronized block so no IO operations nor prolonged obstruction should happen in it, or synchronize the cache access ourselves with ReentrantLock